### PR TITLE
Fix/issue 9648

### PR DESCRIPTION
**Pull request by Khushboo-dev-cpp**
### What does the PR do

1. Moving loading logic to show loading items to nim
2. Error handling for assets also moved to nim
3. when a new account is added, loading state is only shown for that accoun

### Affected areas

Wallet 

### StatusQ checklist

- [ ] add documentation if necessary (new component, new feature)
- [ ] update sandbox app
  - in case of new component, add new component page
  - in case of new features, add variation to existing component page
  - nice to have: add it to the demo application as well
- [ ] test changes in both light and dark theme?

### Screenshot of functionality (including design for comparison)

- [x] I've checked the design and this PR matches it

Uploading accoutnLoading.mov…

### CODE REVIEW

File: ui/imports/shared/views/AssetsView.qml [hunk 1]
```diff
@@ -33,46 +33,53 @@ Item {
...
-                expression: visibleForNetworkWithPositiveBalance
+                expression: visibleForNetworkWithPositiveBalance || loading
             }
         ]
     }
 
+    Component {
+        id: delegateLoader
+        Loader {
+            property variant modelData: model
```
Comment (caybro) [inline ui/imports/shared/views/AssetsView.qml:53]: `variant` is deprecated; just use `var`

File: ui/imports/shared/controls/TokenDelegate.qml [hunk 2]
```diff
@@ -17,16 +17,16 @@ StatusListItem {
     property alias change24HourPercentage: change24HourPercentageText
 
     property string currentCurrencySymbol
-    property string textColor: changePct24hour === undefined  ? Theme.palette.baseColor1 :
-                               Math.sign(changePct24hour) === 0 ? Theme.palette.baseColor1 :
-                               Math.sign(changePct24hour) === -1 ? Theme.palette.dangerColor1 :
+    property string textColor: modelData.changePct24hour === undefined  ? Theme.palette.baseColor1 :
+                               Math.sign(modelData.changePct24hour) === 0 ? Theme.palette.baseColor1 :
```
Comment (caybro) [inline ui/imports/shared/controls/TokenDelegate.qml:21]: `Math.sign()` not needed; I have a déjàvu ;)

File: src/app/modules/shared_models/token_model.nim [hunk 3]
```diff
@@ -172,7 +176,7 @@ QtObject:
       of "change24hour": result = $item.getChange24hour()
       of "currencyPrice": result = $item.getCurrencyPrice()
       of "decimals": result = $item.getDecimals()
-      of "pegSymbol": result = $item.getPegSymbol()
```
Comment (caybro) [inline src/app/modules/shared_models/token_model.nim:175]: `pegSymbol` no longer needed? (still present elsewhere in the model)

File: ui/imports/shared/controls/TokenDelegate.qml (hunk 2 above)
Comment (Khushboo-dev-cpp) [reply ui/imports/shared/controls/TokenDelegate.qml:21]: hahhaha me too now that you pointed it out!

File: src/app/modules/shared_models/token_model.nim (hunk 3 above)
Comment (Khushboo-dev-cpp) [reply src/app/modules/shared_models/token_model.nim:175]: actually I am not sure what this stands for, I dont see a usage on UI side though, should I remove it?

Comment (dlipicar) [reply src/app/modules/shared_models/token_model.nim:175]: it tells you if a crypto is pegged to some fiat currency (so, for `DAI`, the `pegSymbol` will be `USD`). I used this info when we calculated currency formatting on the client side, but that's all done in `status-go` now so this doesn't get used here anymore. I don't mind if it gets removed or if it gets left in peace :D

Comment (Khushboo-dev-cpp) [reply src/app/modules/shared_models/token_model.nim:175]: I'll remove it then

**Comment by Khushboo-dev-cpp** [comment]
all done @caybro

**Review by alaibe — APPROVED** [review]

**Review by caybro — APPROVED** [review]
LGTM!

File: src/app_service/service/wallet_account/service.nim [hunk 4]
```diff
@@ -555,6 +553,12 @@ QtObject:
             tokens = map(tokensDetailsObj.getElems(), proc(x: JsonNode): WalletTokenDto = x.toWalletTokenDto())
             tokens.sort(priorityTokenCmp)
             data.accountsTokens[wAddress] = tokens
+
+            # set assetsLoading to false once the tokens are loaded
```
Comment (saledjenic) [inline src/app_service/service/wallet_account/service.nim:557]: @Khushboo-dev-cpp please make a private proc for setting assets loading to true/false and use it here and in the line 589 below.

`proc updateAssetsLoadingState(self: Service, loading: bool) =`

**Review by saledjenic — APPROVED** [review]

File: src/app_service/service/wallet_account/service.nim (hunk 4 above)
Comment (Khushboo-dev-cpp) [reply src/app_service/service/wallet_account/service.nim:562]: thanks for pointing that out, done :)

**Pull request merged**